pkg/scheduler/extender/extender.go

```diff
@@ -395,11 +409,11 @@ func getReservationName(pod *coreV1.Pod) string {
 	return namespace + "-" + pod.Name
 }
 
-func (e *Extender) createReservation(ctx context.Context, namespace string, name string, nodes []coreV1.Node,
-	capacities []*genV1.CapacityRequest) error {
+func (e *Extender) createReservation(ctx context.Context, ns, name string,
```

Comment:
We return nil after `createReservation` in `filter` function. Suitable nodes list is empty in this case. We should perform `handleReservation` after it became `RESERVED`
```
Feb 14 14:00:55.826 [INFO] [Extender] [createReservation] [0c2d8c1e-e46f-44bd-887c-226cde6a982d] Wait for capacity reservation 'volume-1563-baremetal-injector' become in status 'RESERVED'
Feb 14 14:00:56.829 [INFO] [Extender] [createReservation] [0c2d8c1e-e46f-44bd-887c-226cde6a982d] Capacity reservation 'volume-1563-baremetal-injector' status is 'RESERVED'
Feb 14 14:00:56.829 [INFO] [Extender] [FilterHandler] [baremetal-injector] [0c2d8c1e-e46f-44bd-887c-226cde6a982d] Construct response. Get 3 nodes in request. Among them suitable nodes count is 0. Filtered out nodes - map[]
```

Reply:
I dont get, please explain

Reply:
In `filter` function we read ACR and check it:
- if `IsNotFound` -> call `createReservation` and return nil, so we will response with empty list and `Filter Request` will be repeated by scheduler
- if found -> we call `handleReservation`, build and return suitable nodes list
In my opinion the goal is to process Filter Request only one time, so we need to call `handleReservation` after ACR is created and in RESERVED

Reply:
The function returns `nil` and `handleReservation` called later then `CR` is in `RESERVED` state.

Reply:
Yes, after `FilterHandler` will be repeated. Why do we need it?

Reply:
AFAIK `handleReservation` called after `CR` in the `RESERVED` state.

Reply:
But we don't call it after `createReservation`
```
	// read reservation
	reservation := &acrcrd.AvailableCapacityReservation{}
	err = e.k8sClient.ReadCR(ctx, reservationName, "", reservation)
	if err != nil {
		if k8serrors.IsNotFound(err) {
			// create new reservation
			if err := e.createReservation(ctx, pod.Namespace, reservationName, nodes, capacities); err != nil {
				// cannot create reservation
				return nil, nil, err
			}
			// not an error - reservation requested
			return nil, nil, nil
		}
		// issue with reading reservation
		return nil, nil, err
	}

	// reservation found
	return e.handleReservation(ctx, reservation, nodes)
```

Reply:
The issue is about wait context timeout, your notes is about logic refactoring it's out of this issue scope.

Reply:
Could you please create Github issue and mention it in TODO in `filter` function?

Reply:
You better understand a problem here could you create an issue?

Reply:
Sure